pkg/chat/model.go

```diff
@@ -0,0 +1,30 @@
+package chat
+
+type Chat struct {
+	ID        uint   `gorm:"primaryKey"`
+	URL       string //The url of the chatting window
+	DoctorID  uint
+	PatientID uint
+}
+
+type Message struct {
+	ID        uint `gorm:"primaryKey"`
+	Chat      Chat //Default foreign key:ID in Chat
+	CreatedAt int
+	Type      uint   //Texts, pictures....
+	Content   string //Use escaped string for PictureURL
+}
+
+/*
+Question catagories
+-------------------------------------------------------------------
+If the patient's message contains some keyword specified in DB,
```

Comment:
Umm, if I do not misunderstand, you search the Keywords for catagories, and get questions through catagory, am I right on this?

Reply:
Yes!!!

Reply:
That way would it be better if you put the relationship as: A many to many association from key to catagory, and a catagory has many questions

Reply:
copy that